### PR TITLE
AST: Cache ProtocolDecl::getInheritedProtocols() [5.0]

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes <rdar://problem/47371754>.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

File: include/swift/AST/Decl.h [hunk 1]
```diff
@@ -514,6 +514,9 @@ class alignas(1 << DeclAlignInBits) Decl {
     /// The stage of the circularity check for this protocol.
     Circularity : 2,
 
+    /// Whether we've computed the inherited protocols list yet.
+    InheritedProtocolsValid : 1,
```
Comment (DougGregor) [inline include/swift/AST/Decl.h]: You need to set this to `false` in the constructor!

**Review by DougGregor — CHANGES_REQUESTED** [review]

**Comment by slavapestov** [comment]
@swift-ci Please test

**Review by DougGregor — APPROVED** [review]

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test

**Review by AnnaZaks — APPROVED** [review]

**Pull request merged**